Tiller should print out the resource kind when logging a resource name

Currently tiller only lists the resource name, but if many resources of different kinds have the same name, it becomes impossible to distinguish what tiller is referring to.

As you can see, I have some resources named with a suffix like `-secret` or `-config`, but not all, and then it becomes difficult to determine if the error is on my configmap, deployment, or something else.

Tiller Logs:
```
2017/03/22 21:13:14 release_server.go:913: Executing pre-upgrade hooks for tec-dev
2017/03/22 21:13:14 release_server.go:940: Hooks complete for pre-upgrade tec-dev
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-hookhandler-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-server-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tectonic-com-deploy-dockercfg
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-worker-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme-license-key-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-license-key-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-config
2017/03/22 21:13:17 client.go:381: generating strategic merge patch for *runtime.Unstructured
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-server-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-worker-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-hookhandler-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-nginx-redirect-domain
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-overlord
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-server
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-hookhandler
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-soy-server
2017/03/22 21:13:17 client.go:381: generating strategic merge patch for *runtime.Unstructured
2017/03/22 21:13:17 client.go:246: error updating the resource tec-dev-nginx-redirect-domain:
	 delete patch type with no merge key defined
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-creme
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-dex-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-soy-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-dex-overlord
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-soy-hookhandler
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-nginx-redirect-domain-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for hookhandler
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme
2017/03/22 21:13:17 release_server.go:324: warning: Upgrade "tec-dev" failed: delete patch type with no merge key defined
2017/03/22 21:13:17 storage.go:53: Updating "tec-dev" (v206) in storage
2017/03/22 21:13:17 storage.go:45: Create release "tec-dev" (v207) in storage
```

--- a/pkg/kube/client.go
+++ b/pkg/kube/client.go
@@ -152,7 +152,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		return "", err
 	}
 	err = perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Doing get for: '%s'", info.Name)
+		log.Printf("Doing get for %s: '%s'", info.Mapping.GroupVersionKind.Kind, info.Name)
 		obj, err := resource.NewHelper(info.Client, info.Mapping).Get(info.Namespace, info.Name, info.Export)
 		if err != nil {
 			return err
@@ -390,7 +390,7 @@ func updateResource(c *Client, target *resource.Info, currentObj runtime.Object,
 		return fmt.Errorf("failed to create patch: %s", err)
 	}
 	if patch == nil {
-		log.Printf("Looks like there are no changes for %s", target.Name)
+		log.Printf("Looks like there are no changes for %s %s", target.Mapping.GroupVersionKind.Kind, target.Name)
 		// This needs to happen to make sure that tiller has the latest info from the API
 		// Otherwise there will be no labels and other functions that use labels will panic
 		if err := target.Get(); err != nil {

--- a/pkg/kube/client.go
+++ b/pkg/kube/client.go
@@ -152,7 +152,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		return "", err
 	}
 	err = perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Doing get for %q: %q", info.Mapping.GroupVersionKind.Kind, info.Name)
+		log.Printf("Doing get for %s: %q", info.Mapping.GroupVersionKind.Kind, info.Name)
 		obj, err := resource.NewHelper(info.Client, info.Mapping).Get(info.Namespace, info.Name, info.Export)
 		if err != nil {
 			return err
@@ -186,7 +186,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		}
 		for _, o := range ot {
 			if err := p.PrintObj(o, buf); err != nil {
-				log.Printf("failed to print object type '%s', object: '%s' :\n %v", t, o, err)
+				log.Printf("failed to print object type %d, object: %d :\n %v", t, o, err)
 				return "", err
 			}
 		}
@@ -233,17 +233,17 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 			}
 
 			kind := info.Mapping.GroupVersionKind.Kind
-			log.Printf("Created a new %s called %s\n", kind, info.Name)
+			log.Printf("Created a new %s called %d\n", kind, info.Name)
 			return nil
 		}
 
 		originalInfo := original.Get(info)
 		if originalInfo == nil {
-			return fmt.Errorf("no resource with the name %s found", info.Name)
+			return fmt.Errorf("no resource with the name %d found", info.Name)
 		}
 
 		if err := updateResource(c, info, originalInfo.Object, recreate); err != nil {
-			log.Printf("error updating the resource %s:\n\t %v", info.Name, err)
+			log.Printf("error updating the resource %d:\n\t %v", info.Name, err)
 			updateErrors = append(updateErrors, err.Error())
 		}
 
@@ -258,9 +258,9 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 	}
 
 	for _, info := range original.Difference(target) {
-		log.Printf("Deleting %s in %s...", info.Name, info.Namespace)
+		log.Printf("Deleting %d in %s...", info.Name, info.Namespace)
 		if err := deleteResource(c, info); err != nil {
-			log.Printf("Failed to delete %s, err: %s", info.Name, err)
+			log.Printf("Failed to delete %d, err: %s", info.Name, err)
 		}
 	}
 	if shouldWait {
@@ -278,7 +278,7 @@ func (c *Client) Delete(namespace string, reader io.Reader) error {
 		return err
 	}
 	return perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Starting delete for %s %s", info.Name, info.Mapping.GroupVersionKind.Kind)
+		log.Printf("Starting delete for %d %s", info.Name, info.Mapping.GroupVersionKind.Kind)
 		err := deleteResource(c, info)
 		return skipIfNotFound(err)
 	})
@@ -350,7 +350,7 @@ func deleteResource(c *Client, info *resource.Info) error {
 		}
 		return err
 	}
-	log.Printf("Using reaper for deleting %s", info.Name)
+	log.Printf("Using reaper for deleting %d", info.Name)
 	return reaper.Stop(info.Namespace, info.Name, 0, nil)
 }
 
@@ -390,7 +390,7 @@ func updateResource(c *Client, target *resource.Info, currentObj runtime.Object,
 		return fmt.Errorf("failed to create patch: %s", err)
 	}
 	if patch == nil {
-		log.Printf("Looks like there are no changes for %q %q", target.Mapping.GroupVersionKind.Kind, target.Name)
+		log.Printf("Looks like there are no changes for %s %q", target.Mapping.GroupVersionKind.Kind, target.Name)
 		// This needs to happen to make sure that tiller has the latest info from the API
 		// Otherwise there will be no labels and other functions that use labels will panic
 		if err := target.Get(); err != nil {

--- a/pkg/kube/client.go
+++ b/pkg/kube/client.go
@@ -186,7 +186,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		}
 		for _, o := range ot {
 			if err := p.PrintObj(o, buf); err != nil {
-				log.Printf("failed to print object type %d, object: %d :\n %v", t, o, err)
+				log.Printf("failed to print object type %s, object: %q :\n %v", t, o, err)
 				return "", err
 			}
 		}
@@ -233,17 +233,17 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 			}
 
 			kind := info.Mapping.GroupVersionKind.Kind
-			log.Printf("Created a new %s called %d\n", kind, info.Name)
+			log.Printf("Created a new %s called %q\n", kind, info.Name)
 			return nil
 		}
 
 		originalInfo := original.Get(info)
 		if originalInfo == nil {
-			return fmt.Errorf("no resource with the name %d found", info.Name)
+			return fmt.Errorf("no resource with the name %q found", info.Name)
 		}
 
 		if err := updateResource(c, info, originalInfo.Object, recreate); err != nil {
-			log.Printf("error updating the resource %d:\n\t %v", info.Name, err)
+			log.Printf("error updating the resource %q:\n\t %v", info.Name, err)
 			updateErrors = append(updateErrors, err.Error())
 		}
 
@@ -258,9 +258,9 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 	}
 
 	for _, info := range original.Difference(target) {
-		log.Printf("Deleting %d in %s...", info.Name, info.Namespace)
+		log.Printf("Deleting %q in %s...", info.Name, info.Namespace)
 		if err := deleteResource(c, info); err != nil {
-			log.Printf("Failed to delete %d, err: %s", info.Name, err)
+			log.Printf("Failed to delete %q, err: %s", info.Name, err)
 		}
 	}
 	if shouldWait {
@@ -278,7 +278,7 @@ func (c *Client) Delete(namespace string, reader io.Reader) error {
 		return err
 	}
 	return perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Starting delete for %d %s", info.Name, info.Mapping.GroupVersionKind.Kind)
+		log.Printf("Starting delete for %q %s", info.Name, info.Mapping.GroupVersionKind.Kind)
 		err := deleteResource(c, info)
 		return skipIfNotFound(err)
 	})
@@ -350,7 +350,7 @@ func deleteResource(c *Client, info *resource.Info) error {
 		}
 		return err
 	}
-	log.Printf("Using reaper for deleting %d", info.Name)
+	log.Printf("Using reaper for deleting %q", info.Name)
 	return reaper.Stop(info.Namespace, info.Name, 0, nil)
 }
 

--- a/pkg/kube/client.go
+++ b/pkg/kube/client.go
@@ -152,7 +152,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		return "", err
 	}
 	err = perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Doing get for %s: '%s'", info.Mapping.GroupVersionKind.Kind, info.Name)
+		log.Printf("Doing get for %q: %q", info.Mapping.GroupVersionKind.Kind, info.Name)
 		obj, err := resource.NewHelper(info.Client, info.Mapping).Get(info.Namespace, info.Name, info.Export)
 		if err != nil {
 			return err
@@ -390,7 +390,7 @@ func updateResource(c *Client, target *resource.Info, currentObj runtime.Object,
 		return fmt.Errorf("failed to create patch: %s", err)
 	}
 	if patch == nil {
-		log.Printf("Looks like there are no changes for %s %s", target.Mapping.GroupVersionKind.Kind, target.Name)
+		log.Printf("Looks like there are no changes for %q %q", target.Mapping.GroupVersionKind.Kind, target.Name)
 		// This needs to happen to make sure that tiller has the latest info from the API
 		// Otherwise there will be no labels and other functions that use labels will panic
 		if err := target.Get(); err != nil {

